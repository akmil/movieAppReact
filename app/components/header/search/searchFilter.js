import React from 'react';

export const SearchFilter = ({name}) => {
    return (
        <div className='search-filter-box'>
            Search by
            <ul className={'search-filter-list'}>
                <li>{`Title ${name}`}</li>
                <li>{`Genre ${name}`}</li>
            </ul>
        </div>
    );
};
