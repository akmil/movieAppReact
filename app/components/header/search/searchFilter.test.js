import React from 'react';
import { shallow } from 'enzyme';
import { SearchFilter } from './searchFilter';
import renderer from 'react-test-renderer';

describe('<search SearchFilter/>', () => {
  it('should render <SearchFilter /> components', () => {
    // const wrapper = shallow(<SearchFilter />);
    const wrapper = renderer.create(<SearchFilter/>);
    expect(wrapper).toMatchSnapshot();
  });


  /*
    it('should render an `.icon-star`', () => {
        const wrapper = shallow(<SearchFilter />);
        expect(wrapper.find('.icon-star')).to.have.length(1);
    });

    it('should render children when passed in', () => {
        const wrapper = shallow((
            <SearchFilter>
                <div className="unique" />
            </SearchFilter>
        ));
        expect(wrapper.contains(<div className="unique" />)).to.equal(true);
    }); */

  /* it('simulates click events', () => {
        const onButtonClick = sinon.spy();
        const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
        wrapper.find('button').simulate('click');
        expect(onButtonClick.calledOnce).to.equal(true);
    }); */
});
