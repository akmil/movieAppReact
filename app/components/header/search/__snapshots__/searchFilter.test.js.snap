// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<search /> should render <SearchFilter /> components 1`] = `
<div
  className="search-filter-box"
>
  Search by
  <ul
    className="search-filter-list"
  >
    <li>
      Title undefined
    </li>
    <li>
      Genre undefined
    </li>
  </ul>
</div>
`;

exports[`<search SearchFilter/> should render <SearchFilter /> components 1`] = `
<div
  className="search-filter-box"
>
  Search by
  <ul
    className="search-filter-list"
  >
    <li>
      Title undefined
    </li>
    <li>
      Genre undefined
    </li>
  </ul>
</div>
`;
